compiler/rustc_ast_lowering/src/expr.rs

```diff
@@ -502,16 +515,16 @@ impl<'hir> LoweringContext<'_, 'hir> {
         let if_expr = self.expr(span, if_kind);
         let block = self.block_expr(self.arena.alloc(if_expr));
         let span = self.lower_span(span.with_hi(cond.span.hi()));
-        let opt_label = self.lower_label(opt_label);
```

Comment:
Why?

Reply:
It's the limitation that the recording of label hir id is coupled with `lower_label` method.
So I have to lower label outside otherwise I need to introduce `node_id` and `hir_id` parameters to `lower_expr_while_in_loop_scope`.